### PR TITLE
doc: releases: migration-guide-4.1: Add device driver model update

### PR DESCRIPTION
**Pull request by pdgendt**
Add the list of device drivers APIs that were placed into iterable sections.

Example here: https://builds.zephyrproject.io/zephyr/pr/82333/docs/releases/migration-guide-4.1.html#device-drivers-and-devicetree

### CODE REVIEW

**Review by kartben — COMMENTED** [review]
@pdgendt aren't you hopeful that all driver classes will be migrated and that this can instead just say "[...] for all the upstream driver classes" or something?

**Comment by pdgendt** [comment]
> @pdgendt aren't you hopeful that all driver classes will be migrated and that this can instead just say "[...] for all the upstream driver classes" or something?

Sure, we can simplify, but it does add traceability?

BTW it does show that a lot of driver API structs have no doxygen 😉

**Comment by kartben** [comment]
> BTW it does show that a lot of driver API structs have no doxygen 😉

Yeah but AFAIU it is actually intentional (but definitely not consistently enforced!), since until now the "driver API" has been considered internal and not something we wanted to clutter the public API docs with

**Review by fabiobaltieri — DISMISSED** [review]

File: doc/releases/migration-guide-4.1.rst [hunk 1]
```diff
@@ -57,12 +57,10 @@ LVGL
...
 
-* Device driver APIs are placed into iterable sections (:github:`71773`) to allow for runtime
-  checking. See :ref:`device_driver_api` for more details.
+* Device driver APIs are placed into iterable sections (:github:`71773` and :github:`82102`) to
+  allow for runtime checking. See :ref:`device_driver_api` for more details.
   The :c:macro:`DEVICE_API()` macro should be used by out-of-tree driver implementations for
-  the following driver classes:
-
-    * :c:struct:`adc_driver_api`
+  all the upstream diver classes.
```
Comment (kartben) [inline doc/releases/migration-guide-4.1.rst:63]: 🤿

```suggestion
  all the upstream driver classes.
```

**Review by kartben — CHANGES_REQUESTED** [review]

File: doc/releases/migration-guide-4.1.rst (hunk 1 above)
Comment (pdgendt) [reply doc/releases/migration-guide-4.1.rst]: 🤦

**Review by fabiobaltieri — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**